File: programmers/.vscode/프린터.cpp
#include <iostream>
#include <vector>
#include <queue>

using namespace std;

bool isEnd[101];

int solution(vector<int> priorities, int location){
    int answer = 1;
    int prev_index = 0;
    for (int i=0;i<priorities.size();i++){
        int max_value = 0;
        int max_index = 0;
        
        for (int j=prev_index;j<priorities.size()+prev_index;j++){
            int index = j % priorities.size();
            if (!isEnd[index] && max_value < priorities[index]){
                max_value = priorities[index];
                max_index = index;
            }
        }
        if (max_index == location) return answer;
        prev_index = (max_index + 1) % priorities.size();
        isEnd[max_index] = true;
        answer++;
    }
}

int main(){
    vector<int> priorities;
    priorities.push_back(3);
    priorities.push_back(3);
    priorities.push_back(4);
    priorities.push_back(2);
    
    int location = 3;
    cout << "answer: " << solution(priorities, location) << "\n";
    return 0;
}

File: programmers/level2/프린터.py
def solution(priorities, location):

    printer_q = [(p, loc) for loc, p in enumerate(priorities)]
    answer = 0

    while printer_q:
        priority, loc = printer_q.pop(0)

        for docs in printer_q:
            if priority < docs[0]:
                printer_q.append((priority, loc))
                break
        else:
            answer += 1
            if loc == location:
                break

    return answer
